### ISSUE TITLE
Rule proposal: no unused assignment

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#13240]
**Please describe what the rule should do:**
Report unused left hand side (LHS) in an assignment.

**What category of rule is this? (place an "X" next to just one item)**

[X] Warns about a potential error (problem)

**Provide 2-3 code examples that this rule will warn about:**

```js
function f(){
  let err = db.doStuff('a')
  if (err) {
    handleError(err)
  }
  err = db.doStuff('b') // Variable `err` is assigned but unused.
  // There are two way to fix the code:
  // 1. Handle the variable `err`.
  // 2. Alternatively omit the assignment.
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
The purpose of this rule is very similar to `no-unused-vars`. This rule is needed because `no-unused-vars` does not handle unused (re)assignments like the example.

This issue can be alternatively implemented as an option of `no-unused-vars`.

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by Raduuu** [comment]
I think it could be a rule named like `no-unused-reassignments` for this or, as you said, as an option of `no-unused-vars`.

**Comment by kaicataldo** [comment]
I have a few questions:

1. Just to be clear, this rule won't warn if the variable is declared in a parent scope, correct?
2. How would this rule handle global assignments that might be used in a different file?

**Comment by aladdin-add** [comment]
Thanks for your interest in improving ESLint. As of 2020, unfortunately, we have decided to [only accept rules related to new ECMAScript features](https://eslint.org/blog/2020/05/changes-to-rules-policies). We prefer that new rules be implemented in plugins, so I'm closing it.

Since ESLint is pluggable and can load custom rules at runtime, it doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
